api/src/pcapi/alembic/versions/20241220T095131_4c3be4ff5274_add_timespan_on_headline_offer.py

```diff
@@ -0,0 +1,59 @@
+"""
+add timespan on headline_offer table
+"""
+
+from alembic import op
+import sqlalchemy as sa
+from sqlalchemy.dialects import postgresql
+
+
+# pre/post deployment: post
+# revision identifiers, used by Alembic.
+revision = "4c3be4ff5274"
+down_revision = "4c53718279e7"
+branch_labels: tuple[str] | None = None
+depends_on: list[str] | None = None
+
+
+def upgrade() -> None:
+    op.add_column("headline_offer", sa.Column("timespan", postgresql.TSRANGE(), nullable=False))
+    op.execute(
+        """
+        ALTER TABLE headline_offer
+            ADD CONSTRAINT exclude_offer_timespan EXCLUDE USING gist (
+                "offerId"
+                WITH
+                    =,
+                    timespan
+                WITH
+                    &&
+            )
+        """
+    )
+    op.execute(
+        """
+        ALTER TABLE headline_offer
+            ADD CONSTRAINT exclude_venue_timespan EXCLUDE USING gist (
+                "venueId"
+                WITH
+                    =,
+                    timespan
+                WITH
+                    &&
+            )
+        """
+    )
+    op.drop_column("headline_offer", "dateUpdated")
```

Comment:
J'ai un doute sur le fait que cette migration puisse bien se passer. 
Ce qu'il faut tester : si l'une des étapes timeout (par exemple le drop column), est-ce que le retry ne va pas être bloqué sur une étape précédente (par exemple `constraint already exists` ou le drop column qui échoue parceque la colonne n'existe pas. 
2 solutions pour ça, soit ajouter des IF NOT EXISTS et si on ne peut pas, couper en différentes migrations